src/StoragelessSession/Http/SessionMiddleware.php

```diff
@@ -234,13 +237,26 @@ private function appendToken(SessionInterface $sessionContainer, Response $respo
             return FigResponseCookies::set($response, $this->getExpirationCookie());
         }
 
-        if (! $sessionContainerChanged) {
-            return $response;
+        if ($sessionContainerChanged || $this->shouldTokenBeRefreshed($token)) {
+            return FigResponseCookies::set($response, $this->getTokenCookie($sessionContainer));
+        }
+
+        return $response;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    private function shouldTokenBeRefreshed(Token $token = null) : bool
+    {
+        if (null === $token) {
+            return false;
         }
 
-        return FigResponseCookies::set($response, $this->getTokenCookie($sessionContainer));
+        return time() >= ($token->getClaim('exp') - $this->refreshTime);
```

Comment:
Maybe we should really base ourselves on the "issuedAt" claim, so the naming doesn't need to change, and it's actually clearer ("we refresh the token every 60 seconds" - quite easy to understand).

```php
return time() >= ($token->getClaim('iat') + $this->refreshTime);
```